src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudV1Spec.groovy

```diff
@@ -449,60 +447,63 @@ class FlowCrudV1Spec extends HealthCheckSpecification {
 
         and: "Make all shorter forward paths not preferable. Shorter reverse paths are still preferable"
         cleanupManager.addAction(RESET_ISLS_COST, {database.resetCosts(topology.isls)})
-        possibleFlowPaths.findAll { it.size() == pathNodeCount }.each {
-            pathHelper.getInvolvedIsls(it).each { database.updateIslCost(it, Integer.MAX_VALUE) }
+        List<Isl> modifiedIsls = possibleFlowPaths.findAll { it.size() == pathNodeCount }.collectMany {
+            pathHelper.getInvolvedIsls(it).each {database.updateIslCost(it, Integer.MAX_VALUE) }
         }
 
         when: "Create a flow"
-        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
-        flowHelper.addFlow(flow)
+        def flow = flowFactory.getRandomV1(srcSwitch, dstSwitch)
 
         then: "The flow is built through one of the long paths"
-        def flowPath = northbound.getFlowPath(flow.id)
-        !(PathHelper.convert(flowPath) in possibleFlowPaths.findAll { it.size() == pathNodeCount })
+        def fullFlowPath = flow.retrieveAllEntityPaths()
+        def forwardIsls = fullFlowPath.flowPath.getInvolvedIsls(true)
+        def reverseIsls = fullFlowPath.flowPath.getInvolvedIsls(false)
+        verifyAll {
+            assert forwardIsls.intersect(modifiedIsls).isEmpty()
+            assert reverseIsls.intersect(modifiedIsls).isEmpty()
```

Comment:
No need to verify both forward and revers isls, because we know that kilda builds symmetric paths.

Reply:
done